chat/src/main/java/com/github/twitch4j/chat/events/channel/IRCMessageEvent.java

```diff
@@ -105,11 +123,25 @@ public IRCMessageEvent(String rawMessage) {
      * @param rawMessage  The raw message.
      * @param botOwnerIds The bot owner ids.
      */
-	public IRCMessageEvent(String rawMessage, Collection<String> botOwnerIds) {
+	public IRCMessageEvent(String rawMessage, Map<String, String> channelIdToChannelName, Map<String, String> channelNameToChannelId, Collection<String> botOwnerIds) {
 		this.rawMessage = rawMessage;
+		this.channelIdToChannelName = channelIdToChannelName;
+		this.channelNameToChannelId = channelNameToChannelId;
 
 		this.parseRawMessage();
 
+        // set channel id
+        if (tags.containsKey("room-id")) {
+            channelId = tags.get("room-id");
+        }
+
+        // provide channel id or name from cache if the event was missing one
+        if (!channelName.isPresent() && channelId != null) {
+            channelName = Optional.ofNullable(channelIdToChannelName.get(channelId));
```

Comment:
Watch out for NPE when the map itself is null (same goes for below branch)

Reply:
The map itself can't be null, it's final and initialized in TwitchChat - so unless we change that i don't think we have to worry about a NPE here.

Reply:
Yeah agreed on that point - my comment was more regarding the no args constructor that assigns null for the maps

Reply:
True, i will remove that constructor as we're not using it anywhere. It would be weird to have it in if it's not usable.